packages/ui-drawer-layout/src/DrawerLayout/DrawerContent/index.tsx

```diff
@@ -37,124 +38,109 @@ import generateComponentTheme from './theme'
 import { propTypes, allowedProps } from './props'
 import type {
   DrawerLayoutContentProps,
-  DrawerLayoutContentState
+  DrawerLayoutContentStyleProps
 } from './props'
 
 /**
 ---
 parent: DrawerLayout
 id: DrawerLayout.Content
 ---
+@tsProps
 **/
 @withStyle(generateStyle, generateComponentTheme)
 @testable()
-class DrawerContent extends Component<
-  DrawerLayoutContentProps,
-  DrawerLayoutContentState
-> {
+class DrawerContent extends Component<DrawerLayoutContentProps> {
   static readonly componentId = 'DrawerLayout.Content'
 
   static locatorAttribute = 'data-drawer-content'
   static propTypes = propTypes
   static allowedProps = allowedProps
   static defaultProps = {
-    children: null,
-    // @ts-expect-error ts-migrate(6133) FIXME: 'el' is declared but its value is never read.
-    contentRef: (el: any) => {},
-    // @ts-expect-error ts-migrate(6133) FIXME: 'node' is declared but its value is never read.
-    onSizeChange: (node: any) => {},
     role: 'region'
   }
 
-  state: DrawerLayoutContentState = {
-    shouldTransition: false
-  }
```

Comment:
Refactored the. whole handling of "shouldTransition" because it used to be a prop that we no longer use (even in V7), and since an earlier refactor the state isn't used anymore either, it is directly passed to makeStyles.